presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -158,4 +197,32 @@ public TaskContext addTaskContext(TaskStateMachine taskStateMachine, Session ses
         taskContexts.add(taskContext);
         return taskContext;
     }
+
+    private ListenableFuture<?> reserveMemoryUnsafe(long bytes)
```

Comment:
this methods doesn't seem to be correct. Rethink it (think about different thread/operator scenarios).